src/lib/components/tabs/QTabs.svelte

```diff
@@ -1,67 +1,103 @@
+  // Update the store when "value" changes programmatically
+  $: qTabStore.update(($store) => {
+    $store.value = value;
 
-    return $active;
+    return $store;
   });
-  setContext("activeTab", activeTabStore);
-  setContext("variant", variant);
+  // Update "value" when the store changes from QTab
+  $: value = $qTabStore.value;
 
-  $: value = $activeTabStore.name;
+  setContext("qTabStore", qTabStore);
 
-  $: classes = createClasses([variant, round && "rounded", hidden && "hidden-indicator"], {
-    component: "q-tabs",
-    userClasses,
-  });
+  $: if ($qTabStore.activeEl !== null) {
+    const {
+      previousEl,
+      activeEl,
+      utils: { position, size },
+      variant: storeVariant,
+    } = $qTabStore;
+    const tabsSize = storeVariant === "vertical" ? qTabs.offsetHeight : qTabs.offsetWidth;
+    const tabSize = size / tabsSize;
 
-  let indicatorWidth = derived(activeTabStore, ($activeTabStore) => {
-    return variant === "primary"
-      ? `calc(${$activeTabStore.size}px + 8px)`
-      : `${$activeTabStore.size}px`;
-  });
+    const child = variant === "primary" ? activeEl.firstElementChild as HTMLDivElement : { offsetLeft: 0, offsetWidth: 0 }
+    const previousChild = variant === "primary" ? previousEl?.firstElementChild as HTMLDivElement : { offsetLeft: 0, offsetWidth: 0 }
+
+    if (previousEl === null) {
+      // Position initial indicator
+      qTabs.style.setProperty("--indicator-size", `${tabSize}`);
+      qTabs.style.setProperty("--indicator-position", `${position}px`);
+    } else {
+      // Position indicator on tab change
+      const comparePositions = movementDirection(previousEl, activeEl);
+
+      let transitionSize;
+      if (comparePositions === "next") {
+        // New tab is after the previous one
+        transitionSize =
+          storeVariant === "vertical"
+            ? activeEl.offsetTop + activeEl.offsetHeight - previousEl.offsetTop
+            : (activeEl.offsetLeft + child.offsetLeft) + (child.offsetWidth || activeEl.offsetWidth) - (previousEl.offsetLeft + previousChild.offsetLeft);
+      } else {
+        // New tab is before the previous one
+        transitionSize =
+          storeVariant === "vertical"
+            ? previousEl.offsetTop + previousEl.offsetHeight - activeEl.offsetTop
+            : (previousEl.offsetLeft + previousChild.offsetLeft) + (previousChild.offsetWidth || previousEl.offsetWidth) - (activeEl.offsetLeft + child.offsetLeft);
+        qTabs.style.setProperty("--indicator-position", `${position}px`);
+      }
+
+      qTabs.style.setProperty("--indicator-size", `${transitionSize / tabsSize}`);
 
-  $: style = createStyles(
-    {
-      "--tab-count": QTabCount || 1,
-      "--indicator-size": $indicatorWidth,
-      "--active-tab-index": $activeTabStore.index - 1,
-      "--indicator-position": $activeTabStore.position,
-    },
-    userStyles
-  );
+      setTimeout(() => {
+        qTabs.style.setProperty("--indicator-position", `${position}px`);
```

Comment:
You could use an enum or hard coded constants instead of strings for these strings of CSS variables, so you won't have typos somewhere.